### ISSUE TITLE
DPE Inspector: Editor crashes when Lua Script component is displayed

### ISSUE THREAD

**Issue by LB-ArturZieba** [o3de/o3de#16451]
**Describe the bug**
If DPE is enabled, then Editor will crash if an entity with *Lua Script* component is viewed in the *Inspector*. It should be noted that no `error.log`/`.dmp` files were generated on Linux for this crash.

`error.log` generated on Windows machine when reproducing this issue:
```
Logged at Wed Jul 26 12:37:56 2023
FileVersion: 0.0.0.0
ProductVersion: 0.0.0.0
LogFile: @log@/Editor.log
ProjectDir: C:\Users\GPC-559\O3DE\Projects\NewProject
Executable: Editor.exe (debug: no
Exception Code: 0xC0000005
Exception Addr: 0x0033:0x00007FFACAEBAC10
Exception Module: 
Exception Description: EXCEPTION_ACCESS_VIOLATION, Attempt to read from address 0x00000727
The memory could not be "read"
Call Stack Trace:
25) 00007FFB33B1DDA7 (KERNELBASE) : UnhandledExceptionFilter
24) 00007FFB330DC888 (InProcessClient64) : (function-name not available)
23) 00007FFB330DC4D1 (InProcessClient64) : (function-name not available)
22) 00007FFB35E554B0 (ntdll) : memset
21) 00007FFB35E3C886 (ntdll) : _C_specific_handler
20) 00007FFB35E523AF (ntdll) : _chkstk
19) 00007FFB35E014B4 (ntdll) : RtlRaiseException
18) 00007FFB35E50EBE (ntdll) : KiUserExceptionDispatcher
17) C:\Builds\o3de\Code\Framework\AzCore\AzCore\std\string\string.h (367) : AZStd::basic_string<char,AZStd::char_traits<char>,AZStd::allocator>::assign
16) C:\Builds\o3de\Code\Framework\AzCore\AzCore\RTTI\ReflectContext.h (393) : AZ::AttributeMemberData<AZStd::basic_string<char,AZStd::char_traits<char>,AZStd::allocator> AZ::ScriptProperty::*>::GetAsDomValue
15) C:\Builds\o3de\Code\Framework\AzFramework\AzFramework\DocumentPropertyEditor\Reflection\LegacyReflectionBridge.cpp (1357) : AZ::Reflection::ReadGenericAttributeToDomValue
14) C:\Builds\o3de\Code\Framework\AzFramework\AzFramework\DocumentPropertyEditor\Reflection\LegacyReflectionBridge.cpp (1012) : `AZ::Reflection::LegacyReflectionInternal::InstanceVisitor::CacheAttributes'::`2'::<lambda_1>::operator()
13) C:\Builds\o3de\Code\Framework\AzFramework\AzFramework\DocumentPropertyEditor\Reflection\LegacyReflectionBridge.cpp (1131) : `AZ::Reflection::LegacyReflectionInternal::InstanceVisitor::CacheAttributes'::`2'::<lambda_2>::operator()
12) C:\Builds\o3de\Code\Framework\AzFramework\AzFramework\DocumentPropertyEditor\Reflection\LegacyReflectionBridge.cpp (1143) : AZ::Reflection::LegacyReflectionInternal::InstanceVisitor::CacheAttributes
11) C:\Builds\o3de\Code\Framework\AzFramework\AzFramework\DocumentPropertyEditor\Reflection\LegacyReflectionBridge.cpp (577) : AZ::Reflection::LegacyReflectionInternal::InstanceVisitor::HandleNodeAttributes
10) C:\Builds\o3de\Code\Framework\AzFramework\AzFramework\DocumentPropertyEditor\Reflection\LegacyReflectionBridge.cpp (700) : AZ::Reflection::LegacyReflectionInternal::InstanceVisitor::BeginNode
 9) C:\Builds\o3de\Code\Framework\AzCore\AzCore\std\function\function_template.h (685) : AZStd::function<bool __cdecl(void *,AZ::Serialize::ClassData const *,AZ::Serialize::ClassElement const *)>::operator()
 8) C:\Builds\o3de\Code\Framework\AzCore\AzCore\Serialization\SerializeContext.cpp (1304) : AZ::SerializeContext::EnumerateInstance
 7) C:\Builds\o3de\Code\Framework\AzFramework\AzFramework\DocumentPropertyEditor\Reflection\LegacyReflectionBridge.cpp (1380) : AZ::Reflection::VisitLegacyInMemoryInstance
 6) C:\Builds\o3de\Code\Framework\AzFramework\AzFramework\DocumentPropertyEditor\ReflectionAdapter.cpp (958) : AZ::DocumentPropertyEditor::ReflectionAdapter::GenerateContents
 5) C:\Builds\o3de\Code\Framework\AzFramework\AzFramework\DocumentPropertyEditor\DocumentAdapter.cpp (38) : AZ::DocumentPropertyEditor::DocumentAdapter::GetContents
 4) C:\Builds\o3de\Code\Framework\AzFramework\AzFramework\DocumentPropertyEditor\FilterAdapter.cpp (200) : AZ::DocumentPropertyEditor::RowFilterAdapter::GenerateContents
 3) C:\Builds\o3de\Code\Framework\AzFramework\AzFramework\DocumentPropertyEditor\DocumentAdapter.cpp (38) : AZ::DocumentPropertyEditor::DocumentAdapter::GetContents
 2) C:\Builds\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\UI\DocumentPropertyEditor\DocumentPropertyEditor.cpp (1686) : AzToolsFramework::DocumentPropertyEditor::HandleReset
 1) C:\Builds\o3de\Code\Framework\AzCore\AzCore\EBus\Event.inl (288) : AZ::Event<>::Signal
```

**Steps to reproduce**
1. Execute `ed_enableDPE 1` command in the *Console*.
2. Create an entity and add *Lua Script* component to it.

**Expected behavior**
Editor remains stable.

**Actual behavior**
Editor crashes.

**Video**

https://github.com/o3de/o3de/assets/86952082/55a14ca9-5dd3-415e-9768-e291b786740a

**Found in Branch**
Development (c9db420)

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
c9db420b566866897e7bc41212439fbc0e6ab018

**Desktop**
 - Device: PC
 - OS: Linux
 - Version: Ubuntu 22.04.2 LTS
 - CPU AMD Ryzen 5 3600X 6-Core Processor
 - GPU AMD Radeon RX 5700 XT
 - Memory 16GB

**Comment by alexmontAmazon** [comment]
Fixed by https://github.com/o3de/o3de/pull/16447

**Comment by LB-ArturZieba** [comment]
Issue verified as fixed on the Development (26e6894) build.
